eZ/Publish/SPI/FieldType/Indexable.php

```diff
@@ -24,6 +24,8 @@
      * @param \eZ\Publish\SPI\Persistence\Content\Type\FieldDefinition $fieldDefinition
      *
      * @return \eZ\Publish\SPI\Search\Field[]
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentValue
```

Comment:
How about introducing more specific exception (something like `IndexingExcetption`/`CorruptedDataException` ) with the more precise location of the error (id, version, language code)?

Reply:
I'm totally okay with the idea of introduction of the new specific exception. What do you think @andrerom?

Reply:
I'm ok with it as well, so we can document it on SearchField, then it becomes much more natural to catch it in indexer.

Only question is: naming :P
To identify that, what it the intent here? given the method it is thrown from maybe something like `[Invalid]IndexDataException`

Reply:
For me `InvalidIndexDataException` sounds great 👍 

Where should it be placed?
a) eZ\Publish\Core\Base\Exceptions
b) eZ\Publish\API\Repository\Exceptions
or should we introduce new namespace:
c) eZ\eZ\Publish\SPI\Exceptions

Reply:
+1 for `InvalidIndexDataException` and C

Reply:
I would go for `eZ\Publish\SPI\FieldType\Exceptions\`

Think of SPI folders as potentially separate packages.